File: LessonLockerJPA/src/main/java/com/skilldistillery/lessonlocker/entities/QuizQuestion.java
package com.skilldistillery.lessonlocker.entities;

import java.util.Objects;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "quiz_question")
public class QuizQuestion {
	
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private int id;
	
	// fk quiz_id
	// private Quiz quiz;
	
	// fk question_id
	// private Question question;
	
	public QuizQuestion() {
		super();
	}

	public int getId() {
		return id;
	}

	public void setId(int id) {
		this.id = id;
	}

	@Override
	public int hashCode() {
		return Objects.hash(id);
	}

	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		QuizQuestion other = (QuizQuestion) obj;
		return id == other.id;
	}

	@Override
	public String toString() {
		return "QuizQuestion [id=" + id + "]";
	}


	
}

File: ngLessonLocker/src/app/models/quiz-question.ts
import { Question } from './question';

export class QuizQuestion {
  id: number;
  quizId: number;
  questionId: number;
  question: Question | null;

  constructor(
    id: number = 0,
    quizId: number = 0,
    questionId: number = 0,
    question: Question = new Question()
  ) {
    this.id = id;
    this.quizId = quizId;
    this.questionId = questionId;
    this.question = question;
  }
}
